### PR TITLE
Add "Customizing web app initialization" page

### LINKED ISSUE #7031: Topic request: web initialization
### What information is missing?

Flutter now gives developers more control over the initialization of their web app using the new `flutter.js` and `_flutter.loader` API.  

### How would you like us to fix this problem?

We need to add @ditman's [doc](https://docs.google.com/document/d/1hImwEZvXzDmqhSlt3mSlU3uOTb7T_8GszQPur9P_NEM/edit?resourcekey=0-yyoB2x-amb-37iCDPP6IJw#) somewhere, probably under [development/tools](https://docs.flutter.dev/development/tools) and linked by [/web](https://docs.flutter.dev/web).

### PR DESCRIPTION
**Pull request by johnpryan**
- Moves the "Web FAQ" from "development/platform-integration/web" into a subdirectory "development/platform-integration/web/faq.html"
- Adds "Customizing web app initialization" page to "development/platform-integration/web/initialization"
- Moves "Web renderers" page from `/development/tools` to `development/platform-integration/web`.

CC: @kevmoo @ditman @sfshaza2 

Fixes #7031

### CODE REVIEW

File: src/development/platform-integration/web/initialization.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+</html>
+```
+
+
+
+{{site.alert.note}}
+*Note:* In Flutter 2.10 or earlier,
+this script doesn't support customization.
+To upgrade your index.html file to the latest version,
+see Upgrading an older project.
```
Comment (domesticmouse) [inline src/development/platform-integration/web/initialization.md:69]: Please add a link to the call to action here.

File: src/development/platform-integration/web/initialization.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+{{site.alert.end}}
```
Comment (parlough) [inline src/development/platform-integration/web/initialization.md:70]: - Removes extra "Note" which already exists with `site.alert.note`
- Adds code font to `index.html` to be consistent with rest of file
- Adds link to mentioned section

```suggestion
{{site.alert.note}}
  In Flutter 2.10 or earlier,
  this script doesn't support customization.
  To upgrade your `index.html` file to the latest version,
  see [Upgrading an older project](#upgrading-an-older-project).
{{site.alert.end}}
```

File: src/development/platform-integration/web/initialization.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+If your project was created in Flutter 2.10 or earlier,
+you can create a new `index.html` file
+with the latest initialization template by running `flutter create`.
+
+From your project directory:
+
+```
+flutter create .
+```
```
Comment (parlough) [inline src/development/platform-integration/web/initialization.md:175]:
````suggestion
```
$ flutter create .
```
````

File: src/development/platform-integration/web/initialization.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+## Example: How to display a progress indicator
+
+To give the user of your application feedback
+during the initialization process,
+use the hooks provided for each stage to update the DOM:
+
+
+```
```
Comment (parlough) [inline src/development/platform-integration/web/initialization.md:130]:
````suggestion
```html
````

File: src/development/platform-integration/web/initialization.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+### Initializing the engine
+
+Instead of calling `initializeEngine()` on the engine initializer,
+you can call `autoStart()` to immediately start the app
+with the default configuration
+instead of using the app runner to call `runApp()`:
+
+
+```
```
Comment (parlough) [inline src/development/platform-integration/web/initialization.md:113]:
````suggestion
```js
````

File: src/development/platform-integration/web/initialization.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+at each stage of the initialization process.
+
+## Getting started
+
+By default, the `index.html` file
+generated by the `flutter create` command
+contains a script tag
+that calls `loadEntrypoint` from the `flutter.js` file:
+
+```
```
Comment (parlough) [inline src/development/platform-integration/web/initialization.md:36]:
````suggestion
```html
````

File: src/development/platform-integration/web/initialization.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
+---
+title: Customizing web app initialization
+description: Customize how Flutter apps are initialized on the web
+---
+
+# Customizing web app initialization
+
```
Comment (parlough) [inline src/development/platform-integration/web/initialization.md:7]: Won't this title already be generated?

```suggestion
```

File: src/development/platform-integration/web/initialization.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+* **Loading the entrypoint script** - 
+    Fetches the `main.dart.js` script
+    and initializes the service worker.
+* **Initializing the Flutter engine** -
+    Initializes Flutter's web engine
+    by downloading required resources
+    such as assets, fonts, and CanvasKit.
+* **Running the app** -
+    Prepares the DOM for your Flutter app and runs it.
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:22]: Converting this to a definition list.

```suggestion
**Loading the entrypoint script**
:  Fetches the `main.dart.js` script and initializes the service worker.
**Initializing the Flutter engine**
: Initializes Flutter's web engine by downloading required resources
  such as assets, fonts, and CanvasKit.
**Running the app**
:  Prepares the DOM for your Flutter app and runs it.
```

File: src/development/platform-integration/web/initialization.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+This page shows how to add custom behavior
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:24]:
```suggestion
This page shows how to customize the behavior
```

File: src/development/platform-integration/web/initialization.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+</html>
+```
+
+
+
+{{site.alert.note}}
+*Note:* In Flutter 2.10 or earlier,
+this script doesn't support customization.
+To upgrade your index.html file to the latest version,
+see Upgrading an older project.
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md]:
```suggestion
  In Flutter 2.10 or earlier,
  this script doesn't support customization.
  To upgrade your index.html file to the latest version,
  see [Upgrading an older project](#upgrading).
```

File: src/development/platform-integration/web/initialization.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+{{site.alert.note}}
+  In Flutter 2.10 or earlier,
+  this script doesn't support customization.
+  To upgrade your `index.html` file to the latest version,
+  see [Upgrading an older project](#upgrading-an-older-project).
+{{site.alert.end}}
+
+
+
+The `loadEntrypoint` function returns a JavaScript [Promise][js-promise]
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:72]:
```suggestion
The `loadEntrypoint` function returns a JavaScript [`Promise`][js-promise]
```

File: src/development/platform-integration/web/renderers.md [hunk 12]
```diff
@@ -24,8 +24,8 @@ The `--web-renderer` command line option takes one of three values, `auto`,
 * `auto` (default) - automatically chooses which renderer to use. This option
   chooses the HTML renderer when the app is running in a mobile browser, and
   CanvasKit renderer when the app is running in a desktop browser.
-* `html` - always use the HTML renderer. 
-* `canvaskit` - always use the CanvasKit renderer. 
+* `html` - always use the HTML renderer.
+* `canvaskit` - always use the CanvasKit renderer.
```
Comment (sfshaza2) [inline src/development/platform-integration/web/renderers.md:28]:
```suggestion
* `html` - always use the HTML renderer
* `canvaskit` - always use the CanvasKit renderer
```

File: src/development/platform-integration/web/renderers.md [hunk 13]
```diff
@@ -8,13 +8,13 @@ renderers. This page describes both renderers and how to choose the best one for
...
+: Uses a combination of HTML elements, CSS, Canvas elements, and SVG elements.
+This renderer has a smaller download size.
 
 **CanvasKit renderer**
-  : This renderer is fully consistent with Flutter mobile and desktop, has
-  faster performance with higher widget density, but adds about 2MB in 
-  download size.
+: This renderer is fully consistent with Flutter mobile and desktop, has
+faster performance with higher widget density, but adds about 2MB in
+download size.
```
Comment (sfshaza2) [inline src/development/platform-integration/web/renderers.md:17]:
```suggestion
: This renderer is fully consistent with Flutter mobile and desktop, has
  faster performance with higher widget density, but adds about 2MB in
  download size.
```

File: src/development/platform-integration/web/renderers.md [hunk 14]
```diff
@@ -8,13 +8,13 @@ renderers. This page describes both renderers and how to choose the best one for
 your needs. The two renderers are:
 
 **HTML renderer**
-  : Uses a combination of HTML elements, CSS, Canvas elements, and SVG elements.
-  This renderer has a smaller download size.
+: Uses a combination of HTML elements, CSS, Canvas elements, and SVG elements.
+This renderer has a smaller download size.
```
Comment (sfshaza2) [inline src/development/platform-integration/web/renderers.md:12]:
```suggestion
: Uses a combination of HTML elements, CSS, Canvas elements, and SVG elements.
  This renderer has a smaller download size.
```

File: src/development/platform-integration/web/initialization.md [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+      </script>
+   </body>
+</html>
+```
+
+
+For a more practical example using CSS animations,
+see the [initialization code][gallery-init] for the Flutter Gallery.
+
+[gallery-init]: https://github.com/flutter/gallery/blob/master/web/index.html
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:161]:
```suggestion
[gallery-init]: {{site.github}}/flutter/gallery/blob/master/web/index.html
```

File: src/development/platform-integration/web/initialization.md [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+
+## Upgrading an older project
+
+If your project was created in Flutter 2.10 or earlier,
+you can create a new `index.html` file
+with the latest initialization template by running `flutter create`.
+
+From your project directory:
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:169]:
```suggestion
From your project directory, run the following:
```

File: src/development/platform-integration/web/initialization.md [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+_flutter.loader.loadEntrypoint({
+  serviceWorker: {
+    serviceWorkerVersion: serviceWorkerVersion,
+  }
+}).then(function(engineInitializer) {
+  return engineInitializer.autoStart();
+});
+```
+
+## Example: How to display a progress indicator
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:121]:
```suggestion
## Example: Display a progress indicator
```

File: src/development/platform-integration/web/index.md [hunk 18]
```diff
@@ -0,0 +1,4 @@
+---
+layout: toc
+title: Web
```
Comment (sfshaza2) [inline src/development/platform-integration/web/index.md:3]:
```suggestion
title: Web
layout: toc
```

File: src/development/platform-integration/web/initialization.md [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+The `loadEntrypoint` method accepts these parameters:
+
+* `entrypointUrl`: The URL of your Flutter app's entrypoint.
+      Defaults to `main.dart.js`.
+* `serviceWorker`: The configuration of the `flutter_service_worker.js`.
+      If this isn’t set, the service worker won’t be used.
+    * `serviceWorkerVersion`: pass *the `serviceWorkerVersion` var* set by
+        the build process in your <strong><code>index.html</code></strong> file
+    * `timeoutMillis`: the timeout value for the service worker load.
+        Defaults to <strong><code>4000ms</code></strong>.
```
Comment (sfshaza2) [inline src/development/platform-integration/web/initialization.md:100]:
```suggestion
`entrypointUrl`
: The URL of your Flutter app's entrypoint. Defaults to `main.dart.js`.

`serviceWorker`
: The configuration of the `flutter_service_worker.js` file.
  If this isn’t set, the service worker won’t be used.
      
`serviceWorkerVersion`
: Pass *the `serviceWorkerVersion` var* set by
  the build process in your <strong><code>index.html</code></strong> file.
  
`timeoutMillis`
: The timeout value for the service worker load.
  Defaults to <strong><code>4000ms</code></strong>.
```

**Review by sfshaza2 — APPROVED** [review]
lgtm with some minor edits

File: src/_data/sidenav.yml [hunk 20]
```diff
@@ -274,8 +281,6 @@
           permalink: /development/tools/flutter-fix
         - title: Code formatting
           permalink: /development/tools/formatting
-        - title: Web renderers
-          permalink: /development/tools/web-renderers
```
Comment (kevmoo) [inline src/_data/sidenav.yml:278]: Should we have redirects added for the old page?

Comment (sfshaza2) [reply src/_data/sidenav.yml:278]: Yes, I was going to ask John for that. :D

**Comment by sfshaza2** [comment]
@johnpryan, I'm going to land this, but can you create a PR to set up redirects for the moved files?

**Pull request merged**